pkg/types/types.go

```diff
@@ -197,29 +197,59 @@ type Metric struct {
 }
 
 // MergeMetrics merges metrics by name.
-func MergeMetrics(metrics [][]Metric) []Metric {
+// It returns merged metrics, number of rendered data points for the returned metrics,
+// and number of inconsistent data points seen (if consistencyCheck is true).
+func MergeMetrics(metrics [][]Metric, consistencyCheck bool) ([]Metric, int, int) {
 	if len(metrics) == 0 {
-		return nil
+		return nil, 0, 0
 	}
 
 	if len(metrics) == 1 {
-		return metrics[0]
+		pointCount := 0
+		ms := metrics[0]
+		for _, m := range ms {
+			pointCount += len(m.Values)
+		}
+		return ms, pointCount, 0
 	}
 
-	names := make(map[string][]Metric)
+	metricByNames := make(map[string][]Metric)
 
 	for _, ms := range metrics {
 		for _, m := range ms {
-			names[m.Name] = append(names[m.Name], m)
+			metricByNames[m.Name] = append(metricByNames[m.Name], m)
 		}
 	}
 
 	merged := make([]Metric, 0)
-	for _, ms := range names {
-		merged = append(merged, mergeMetrics(ms))
+	pointCount := 0
+	inconsistencyCount := 0
+	inconsMetricReportLimit := 10
+	var inconsMetricReport []map[string]interface{}
```

Comment:
Sorry @emadolsky but `interface{}` is not ok here as it is not in most places) Please replace with a struct if necessary.
Also, isn't `metricName` unique? If yes, we can simply use a `map[string]struct`.
Also, aren't `start` and `stop` the same for all metrics? If yes, we can put it on top level.

If answers are yes to last 2 questions, we can simply go with `map[string]int` as `step` is not really important.

Reply:
> Also, isn't metricName unique? If yes, we can simply use a map[string]struct.

I think it is more suitable to have predefined keys and differing values in our log schema. This is the main reason that I did not use metricName as the key.

> Also, aren't start and stop the same for all metrics? If yes, we can put it on top level.

I was not sure that they would be the same for all the metrics returned in a render request. So if you verify that it is always the case, you are absolutely right.

Reply:
> but interface{} is not ok here as it is not in most places

I wasn't quite happy about using it too. But unfortunately I observed that `zap.Any` doesn't work as expected on structs. I can convert all the values to strings, but I prefer not to do that.

Reply:
It's up to you how you do it, but `interface{}` is a red flag unless there's no other choice (like parsing schema-less data, etc.) Please replace it with something else. Again, it's not critical with what specifically you replace it with as long as there's no `interface{}`. I can give you more detail in a conversation.

Reply:
used struct with json tags. lgty?